kubeconfig.go

```diff
@@ -34,16 +42,26 @@ func New(config Config) (*KubeConfig, error) {
 	}
 
 	g := &KubeConfig{
+		g8sClient: config.G8sClient,
 		logger:    config.Logger,
 		k8sClient: config.K8sClient,
 	}
 
 	return g, nil
 }
 
-// NewG8sClientFromSecret returns a generated clientset based on the kubeconfig stored in a secret.
-func (k KubeConfig) NewG8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (versioned.Interface, error) {
-	kubeConfig, err := k.getKubeConfigFromSecret(ctx, secretName, secretNamespace)
+// NewG8sClientForApp returns a generated clientset for the cluster configured
+// in the kubeconfig section of the app CR. If this is empty a clientset for
+// the current cluster is returned.
+func (k KubeConfig) NewG8sClientForApp(ctx context.Context, app v1alpha1.App) (versioned.Interface, error) {
+	secretName := secretName(app)
+
+	// KubeConfig is not configured so connect to current cluster.
+	if secretName == "" {
+		return k.g8sClient, nil
+	}
```

Comment:
This is needed because app-operator needs to support control plane clusters for VOO and tenant clusters for App Catalog.